week-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,24 @@
-function setAlarm() {}
+function setAlarm() {
+  const input = document.getElementById("alarmSet");
+  const heading = document.getElementById("timeRemaining");
+  let time = Number(input.value);
+
+  function updateHeading() {
+    const minutes = Math.floor(time / 60);
+    const seconds = time % 60;
+    heading.innerText = `Time Remaining: ${minutes
+      .toString()
+      .padStart(2, "0")}:${seconds.toString().padStart(2, "0")}`;
+
+    if (time === 0) {
+      playAlarm();
+      clearInterval(intervalId);
+    }
+    time = time - 1;
+  }
```

Comment:
Would you be able to try to refactor it a bit? Try to declare`updateHeading` function outside of `setAlarm` function? It is a good practice to declare functions separately if possible.